packages/core/saved-objects/core-saved-objects-server/src/extensions/security.ts

```diff
@@ -6,24 +6,187 @@
  * Side Public License, v 1.
  */
 
-import type { SavedObject } from '@kbn/core-saved-objects-common';
-import type { EcsEvent } from '@kbn/ecs';
+import {
+  SavedObjectReferenceWithContext,
+  SavedObjectsFindResult,
+  SavedObjectsResolveResponse,
+} from '@kbn/core-saved-objects-api-server';
+import type { BulkResolveError, LegacyUrlAliasTarget } from '@kbn/core-saved-objects-common';
+import { EcsEvent } from '@kbn/ecs';
+import { SavedObject } from '../..';
 
 /**
- * The PerformAuthorizationParams interface contains settings for checking
- * & enforcing authorization via the ISavedObjectsSecurityExtension.
+ * The SecurityAction enumeration contains values for all
+ * valid shared object security actions. The string for
+ * each value correlates to the ES operation.
  */
-export interface PerformAuthorizationParams<A extends string> {
+export enum SecurityAction {
+  CHECK_CONFLICTS,
+  CLOSE_POINT_IN_TIME,
+  COLLECT_MULTINAMESPACE_REFERENCES,
+  COLLECT_MULTINAMESPACE_REFERENCES_UPDATE_SPACES,
+  CREATE,
+  BULK_CREATE,
```

Comment:
Are all the types in this file still used by core? Given most of the logic moved back to the security extension's implementation, I suspect some of them may not?

Reply:
Thanks for catching this! It was originally put here when I was attempting a single catch-all authz method and it doesn't need to be there anymore.